Integration with Sonar

bsodzik opened the pull request:
Hello there,

First of all, thank you for your work on mocha-jenkins-reporter. I find colored console output especially charming :)

In this pull request I would like to propose a little addition to your reporter which can extend its capabilities significantly. It is integration with Sonar, a quite popular tool responsible for static code analysis in continuous integration process. Unfortunately, the output generated by your reporter is not compatible with Sonar expectations. The only required change is to generate relative path to a test file and assign it to classname property.

There is already a reporter for Sonar (https://github.com/pghalliday/mocha-sonar-reporter) but first of all the project is not very active and secondly there is no support for colored console output :)

Please take a look at my commit and decide if it can be merged with your repository. If you find it worth then I will add some documentation for that feature.

Kind regards,
Tomasz

jvah commented:
First of all, sorry for replying late. For some weird reason I don't get email notifications of this repository and don't always remember to check for new issues, should really find out what's the issue with GitHub.

Anyway, the PR looks quite good, thank you! Do you think you could document the `JENKINS_REPORTER_ENABLE_SONAR` properly in the README.md so that people know about that option. Also the `relativeFilePath.replace('.js', '');` feels a bit dirty, would it work if that would be changed into `relativeFilePath.replace(/\.js$/, '');` instead? Then it would only match the end of the path. Other than these two minor details I can merge this right away.

bsodzik commented:
Thanks for replying, I was losing any hope ;)

When it comes to this `replace` I agree it is a little bit ugly. I was thinking about .slice(0, -3), but it is even worse. I changed it according to your suggestion, it should work well.

I also added some documentation. Unfortunately, I am not working on the project where I had Sonar and Jenkins so I can not provide full blown documentation on how to integrate all those tools. On the other hand, this is probably not a place for such a detailed doc.

At the end, I completely screwed this pull request because I forgot to add `--rebase` when pulling your last changes. Please ignore this one and instead pull https://github.com/futurice/mocha-jenkins-reporter/pull/18. Sorry for mess :(
